dataset_builders/pie/cdcp/cdcp.py

```diff
@@ -0,0 +1,159 @@
+    relation_int2str: Callable[[int], str],
+    proposition_int2str: Callable[[int], str],
+):
+    document = CDCPDocument(id=example["id"], text=example["text"])
+    for proposition_dict in dl2ld(example["propositions"]):
+        proposition = LabeledSpan(
+            start=proposition_dict["start"],
+            end=proposition_dict["end"],
+            label=proposition_int2str(proposition_dict["label"]),
+        )
+        document.propositions.append(proposition)
+        if proposition_dict.get("url", "") != "":
+            url = Attribute(annotation=proposition, value=proposition_dict["url"])
+            document.urls.append(url)
+
+    for relation_dict in dl2ld(example["relations"]):
+        relation = BinaryRelation(
+            head=document.propositions[relation_dict["head"]],
+            tail=document.propositions[relation_dict["tail"]],
+            label=relation_int2str(relation_dict["label"]),
+        )
+        document.relations.append(relation)
+
+    return document
+
+
+def document_to_example(
+    document: CDCPDocument,
+    relation_str2int: Callable[[str], int],
+    proposition_str2int: Callable[[str], int],
+) -> Dict[str, Any]:
+    result = {"id": document.id, "text": document.text}
+    proposition2dict = {}
+    proposition2idx = {}
+    for idx, proposition in enumerate(document.propositions):
+        proposition2dict[proposition] = {
+            "start": proposition.start,
+            "end": proposition.end,
+            "label": proposition_str2int(proposition.label),
+            "url": "",
+        }
+        proposition2idx[proposition] = idx
+    for url in document.urls:
+        proposition2dict[url.annotation]["url"] = url.value
+
+    result["propositions"] = ld2dl(
+        proposition2dict.values(), keys=["start", "end", "label", "url"]
+    )
+
+    relations = [
+        {
+            "head": proposition2idx[relation.head],
+            "tail": proposition2idx[relation.tail],
+            "label": relation_str2int(relation.label),
+        }
+        for relation in document.relations
+    ]
+    result["relations"] = ld2dl(relations, keys=["head", "tail", "label"])
+
+    return result
+
+
+def convert_to_text_document_with_labeled_spans_and_binary_relations(
+    document: CDCPDocument,
+    verbose: bool = True,
+) -> TextDocumentWithLabeledSpansAndBinaryRelations:
+    doc_simplified = document.as_type(
+        TextDocumentWithLabeledSpansAndBinaryRelations,
+        field_mapping={"propositions": "labeled_spans", "relations": "binary_relations"},
+    )
+    result = trim_text_spans(
+        doc_simplified,
+        layer="labeled_spans",
+        verbose=verbose,
+    )
+    return result
+
+
+class CDCPConfig(datasets.BuilderConfig):
+    """BuilderConfig for CDCP."""
+
+    def __init__(self, **kwargs):
+        """BuilderConfig for CDCP.
+
+        Args:
+          **kwargs: keyword arguments forwarded to super.
+        """
+        super().__init__(**kwargs)

```

Comment:
remove this class at all, just use `datasets.BuilderConfig` directly

Reply:
fix in 0a66d06